content/blog/2020/08/2020-08-24-custom-distribution-service.adoc

```diff
@@ -0,0 +1,234 @@
+implement the `Error Controller` interface and include a custom message in it. 
+This was technical debt from the last phase and was completed and merged during this phase.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/92[#92]
+
+=== PMD Analysis
+
+This was suggested by Martin which was a real life-saver. The project originally did not 
+include PMD and was added this phase. It helped me catch tons of errors. When the initial 
+PMD check was run and we found approximately 162 PMD errors. We realised some of them were not
+relevant and some of them could be fixed later. 
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/102[#102]
+
+=== Findbugs Analysis
+
+Another tool to improve code quality that we included in this phase was findbugs. 
+It did catch around 5-10 bugs in my code which I immediately resolved. Most of them were 
+around the `Closeable HTTP Request` and an easy fix was the `try with resources`.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/118[#118]
+
+=== Jacoco Code Coverage
+
+We needed to make sure most of the code we write had proper coverage for all branches and
+lines. Therefore we decided to include a JaCoco Code Coverage reporter that helped us find the
+uncovered lines and areas we need to improve coverage on.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/103[#103]
+
+=== Remove JCasC generation
+
+While developing the service we quickly realised that the generation of the war package broke if we 
+included a `configuration as code` section but did not provide a path to the corresponding required `yml`
+file. Therefore we took a decision to remove the `casc` section all together. Maybe it will comeback in
+a future patch
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/127[#127]
+
+==== Minor Fixes 
+
+** Logging Fix: link:https://github.com/jenkinsci/custom-distribution-service/pull/99[#99]
+** Docs Fix : link: https://github.com/jenkinsci/custom-distribution-service/pull/120[#120]
+** Update Center Dump Fix : link: https://github.com/jenkinsci/custom-distribution-service/pull/125[#125]
+** Class Path Fix: link: https://github.com/jenkinsci/custom-distribution-service/pull/126[#126]
+** Release Drafter Addition: link: https://github.com/jenkinsci/custom-distribution-service/pull/136[#136]
+
+== Front end
+
+=== Community Config Navigation link
+
+There was no community configuration link present for navigation which was added here.
+ Now it is easier to navigate to the community page from the home page itself.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/100[#100]
+
+== Docker updates
+
+=== Build everything with Docker
+
+This was one of the major changes with respect to docker this phase. Initially the process was
```

Comment:
It was so much more than docker!  You can 100% highlight here how this change makes it easy for community adoption removing the requirement to have to build everything locally.

Reply:
Yeah : )